### PR TITLE
Attempt multiple different token requests

### PR DESCRIPTION
**Pull request by arjunrn**
The token request with client header authorization does not encode the header client identifier payload with URLEscape like the other request methods. The current oauth-server implements it so but this behavior has changed in newer versions of the dependency. So now the test which makes the token requests can multiple requests to fetch the token and as long as one of them succeeds the test is considered as succeeded.

### CODE REVIEW

File: test/extended/oauth/client_secret.go [hunk 1]
```diff
@@ -14,13 +14,13 @@ import (
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
```
Comment (slaskawi) [inline test/extended/oauth/client_secret.go:17]: Please leave the imports where they were. This change is not required.

**Review by slaskawi — CHANGES_REQUESTED** [review]
Only one nitpick. Otherwise looks good.

**Review by slaskawi — APPROVED** [review]
/lgtm
/approve

File: test/extended/oauth/client_secret.go [hunk 2]
```diff
@@ -136,17 +148,24 @@ func bodyClientAuthRequest(tokenEndpoint string) *http.Request {
...
-func headerClientAuthRequest(tokenEndpoint string) *http.Request {
-	req, err := http.NewRequest("POST", tokenEndpoint,
+func headerClientAuthRequest(tokenEndpoint string) []*http.Request {
+	reqNoURLEscape, err := http.NewRequest("POST", tokenEndpoint,
 		bytes.NewBufferString("grant_type=authorization_code&"+
 			"code="+authzToken,
 		),
 	)
 	o.Expect(err).NotTo(o.HaveOccurred())
+	reqNoURLEscape.Header.Set("Authorization", fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte("oauth-client-with-plus:secret+with+plus"))))
```
Comment (stlaz) [inline test/extended/oauth/client_secret.go:161]: Please use https://pkg.go.dev/net/http#Request.SetBasicAuth for setting basic auth

File: test/extended/oauth/client_secret.go [hunk 3]
```diff
@@ -136,17 +148,24 @@ func bodyClientAuthRequest(tokenEndpoint string) *http.Request {
...
 
-	req.Header.Set("Authorization", fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte("oauth-client-with-plus:secret+with+plus"))))
-	return req
+	reqURLEscape, err := http.NewRequest("POST", tokenEndpoint,
+		bytes.NewBufferString("grant_type=authorization_code&"+
+			"code="+authzToken,
+		),
+	)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	reqURLEscape.Header.Set("Authorization", fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte("oauth-client-with-plus:secret%2Bwith%2Bplus"))))
```
Comment (stlaz) [inline test/extended/oauth/client_secret.go:169]: use the library basic auth setter here too, please

**Pull request merged**